installer/bundle/bundle_skel.sh

```diff
@@ -768,6 +844,8 @@ install_extra_package()
 #
 
 ulinux_detect_installer
+install_troubleshooter
```

Comment:
this is still not the best place for this to be, notice if customer is running shell bundle with `--purge` or w/e, you will first try installing the troubleshooter and then it will be removed again. I think the ideal place for this would be only in the execution path for `--install` or `--upgrade`, at the END of the install/upgrade process. That way there is the least risk for interference. You may want to `set +e` before calling and then reset with `set -e` afterwards, so that the script doesn't exit if there is some uncaught exception. Just make sure that, even if it is the last thing that executes, a failure `install_troubleshooter` won't cause the bundle exit code to be nonzero

Reply:
That makes sense, one of my main thoughts though was that I wanted it to try and install at the beginning of the install process, in case OMS install failed the troubleshooter would still be on the machine

Reply:
But as long as it'll install at the end even if the shell bundle exits out then that works better even

Reply:
in that case maybe put at beginning of install or upgrade process
just make sure to `set +e` or w/e method to prevent total failure on tst fail

Reply:
Looking at Dmitry's comment below, adding the troubleshooter install to the end of the install process does seem like a better option (where it will only install via wget if OMS install is set to return a non-0 exit code, since that means the troubleshooter couldn't install through the base_omsagent.data files), so I think you're right in putting it at the end (with the `set +e` and `set -e` bookends around it of course)

Reply:
I updated the installation to occur at the end, and only when `--install` or `--upgrade` are used (see https://github.com/microsoft/OMS-Agent-for-Linux/pull/1190#issuecomment-663785862 for more in-depth analysis), hopefully it shouldn't interfere with OMS installation at all